nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,390 @@
+        final String endpoint = (String)coordinates.get(ENDPOINT_KEY);
+        final String mimeType = (String)coordinates.get(MIME_TYPE_KEY);
+        final String method   = (String)coordinates.get(METHOD_KEY);
+        final String body     = (String)coordinates.get(BODY_KEY);
+
+        if (StringUtils.isBlank(body) && (method.equals("post") || method.equals("put"))) {
+            throw new LookupFailureException(
+                String.format("Used HTTP verb %s without specifying the %s key to provide a payload.", method, BODY_KEY)
+            );
+        }
+
+        Request request = buildRequest(mimeType, method, body, endpoint);
+        try {
+            Response response = executeRequest(request);
+            InputStream is = response.body().byteStream();
+
+            Record record = handleResponse(is, coordinates);
+
+            return record != null
+                    ? Optional.of(record)
+                    : Optional.empty();
+        } catch (Exception e) {
+            getLogger().error("Could not execute lookup.", e);
+            throw new LookupFailureException(e);
+        }
+    }
+
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        if (propertyDescriptorName.startsWith("header")) {
+            String header = propertyDescriptorName.substring(propertyDescriptorName.indexOf(".") + 1);
+            return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .displayName(header)
+                .addValidator(Validator.VALID)
+                .dynamic(true)
+                .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+                .build();
+        }
+
+        return null;
+    }
+
+    protected Response executeRequest(Request request) throws IOException {
+        return client.newCall(request).execute();
+    }
+
+    private Record handleResponse(InputStream is, Map<String, Object> coordinates) throws SchemaNotFoundException, MalformedRecordException, IOException {
+        Map<String, String> variables = coordinates.entrySet().stream()
+            .collect(Collectors.toMap(
+                e -> e.getKey(),
+                e -> e.getValue().toString()
+            ));
+        try (RecordReader reader = readerFactory.createRecordReader(variables, is, getLogger())) {
+
+            Record record = reader.nextRecord();
+
+            if (recordPath != null) {
+                Optional<FieldValue> fv = recordPath.evaluate(record).getSelectedFields().findFirst();
+                if (fv.isPresent()) {
+                    FieldValue fieldValue = fv.get();
+                    RecordSchema schema = new SimpleRecordSchema(Arrays.asList(fieldValue.getField()));
+                    String[] parts = recordPath.getPath().split("/");
+                    String last = parts[parts.length - 1];
```

Comment:
I don't think this is actually what we want here. What if the path is something like `/myField[./second/third = 'hello']` - in that case, `last` will equal `third = 'hello']`

Reply:
Great point. I think I'm going to have to make another property to handle this instead.

Reply:
Done.